### PR TITLE
Database specific interfaces

### PR DESCRIPTION
**Pull request by willsb**
Squash whenever

### CODE REVIEW

File: Toggl.PrimeRadiant/Models/DatabaseInterfaces.tt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+<#@ assembly name="System.Core" #>
+<#@ import namespace="System.IO" #>
+<#@ import namespace="System.Linq" #>
+using Toggl.Multivac.Models;
+
+namespace Toggl.PrimeRadiant.Models
+{
+<#
+    var filePath = Path.GetDirectoryName(Host.TemplateFile);
+    var solutionDirectory = Directory.GetParent(Directory.GetParent(filePath).ToString()).ToString();
```
Comment (amulware) [inline Toggl.PrimeRadiant/Models/DatabaseInterfaces.tt]: "../../" shenanigans might be clearer.

File: Toggl.PrimeRadiant/Models/DatabaseInterfaces.tt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    var modelsToGenerate = Directory.GetFiles(modelsFolder, "*.cs");
+
+    foreach(var modelFile in modelsToGenerate)
+    {
+        var interfaceName = Path.GetFileName(modelFile).Replace(".cs", "");
+        if (interfaceName == "IBaseModel") continue;
+
+        var className = interfaceName.Substring(1, interfaceName.Length - 1);
+
+        Write(string.Format("    public interface IDatabase{0} : {1}, IDatabaseSyncable", className, interfaceName));
```
Comment (amulware) [inline Toggl.PrimeRadiant/Models/DatabaseInterfaces.tt]: Interpolated strings are not supported?

Comment (willsb) [inline Toggl.PrimeRadiant/Models/DatabaseInterfaces.tt]: I am so used at using C# 4 when doing t4 that I didn't even consider 😄

File: Toggl.PrimeRadiant/Models/DatabaseInterfaces.tt [hunk 3]
```diff
@@ -8,9 +8,8 @@ namespace Toggl.PrimeRadiant.Models
 {
 <#
     var filePath = Path.GetDirectoryName(Host.TemplateFile);
-    var solutionDirectory = Directory.GetParent(Directory.GetParent(filePath).ToString()).ToString();
-    var modelsFolder = solutionDirectory + "/Toggl.Multivac/Models/";
-    var modelsToGenerate = Directory.GetFiles(modelsFolder, "*.cs");
+    var solutionDirectory = Path.Combine(filePath, @"../../Toggl.Multivac/Models/" );
```
Comment (amulware) [inline Toggl.PrimeRadiant/Models/DatabaseInterfaces.tt]: :fire: @?

**Pull request merged**